### ISSUE TITLE
Test (task collection and) workflow apply from a user different from `fractal`

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#214]

**Comment by tcompa** [comment]
First part of the test (all executed from a non-fractal user, while the server is alive):

```
$ cat .fractal.env 
FRACTAL_USER=test01@me.com
FRACTAL_PASSWORD=test01
FRACTAL_SERVER=http://localhost:8002
SLURM_USER=test01

$ fractal register -p test01 test01@me.com test01
...

$ fractal task collect fractal-tasks-core
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:root:Namespace(batch=False, cmd='task', json=False, package='fractal-tasks-core', package_extras=None, password=None, python_version=None, slurm_user=None, subcmd='collect', user=None, v=0)
DEBUG:httpx._client:HTTP Request: POST http://localhost:8002/auth/token/login "HTTP/1.1 200 OK"
DEBUG:httpx._client:HTTP Request: POST http://localhost:8002/api/v1/task/collect/pip/ "HTTP/1.1 201 Created"
{
  "package": "fractal-tasks-core",
  "venv_path": ".fractal/fractal-tasks-core",
  "info": "Collecting tasks in the background. GET /task/collect/{venv_path} to query collection status"
}

# WAIT A BIT..

$ fractal task check-collection /data/homes/fractal/fractal-server/fractal-demos/examples/server/FRACTAL_ROOT/.fractal/fractal-tasks-core/
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:root:Namespace(batch=False, cmd='task', installation_path='/data/homes/fractal/fractal-server/fractal-demos/examples/server/FRACTAL_ROOT/.fractal/fractal-tasks-core/', json=False, password=None, slurm_user=None, subcmd='check-collection', user=None, v=0)
DEBUG:httpx._client:HTTP Request: POST http://localhost:8002/auth/token/login "HTTP/1.1 200 OK"
DEBUG:httpx._client:HTTP Request: GET http://localhost:8002/api/v1/task/collect//data/homes/fractal/fractal-server/fractal-demos/examples/server/FRACTAL_ROOT/.fractal/fractal-tasks-core/ "HTTP/1.1 200 OK"
{
  "status": "OK",
  "task_list": [ ... ]
```

And after that, we can do
```
$ /data/homes/fractal/fractal-server/fractal-demos/examples/server/FRACTAL_ROOT/.fractal/fractal-tasks-core/venv/bin/python
Python 3.8.0 (default, Nov  6 2019, 21:49:08) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fractal_tasks_core
```

TL;DR
Task collection can be triggered by a non-admin user, and the resulting env is accessible to them.

**Comment by tcompa** [comment]
To add:
- [ ] Run 50 workflows at the same time and verify that the server does not crash.

**Comment by jluethi** [comment]
As an addition to the note above: Is it feasible to include a test with many different users? Maybe they'll all be the same slurm user. But if they are different fractal users, we may catch some more edge cases

**Comment by tcompa** [comment]
This is not an actual test (as in "it will run on github CI at every push"), but rather something (essentially a script) that we need to run by hand on a specific cluster. Once we have it in a reasonably clean version, we can also include different users.

**Comment by tcompa** [comment]
We successfully tested [this example](https://github.com/fractal-analytics-platform/fractal-demos/tree/0604ad011d100dfbd5373a0b8736ea2ea29d0a83) locally (with the `process` backend).

Tests on SLURM will likely hit permission errors for the logs folder, and then I'll close this issue in favor of new ones.

**Comment by tcompa** [comment]
Ref: https://github.com/fractal-analytics-platform/fractal-server/issues/253

**Comment by tcompa** [comment]
This is now complete, after #259 and as of https://github.com/fractal-analytics-platform/fractal-demos/tree/29c3f6497ed9e5579ba023fade3de29f981e2b37.
